astropy/stats/bayesian_blocks.py

```diff
@@ -342,6 +331,9 @@ def fit(self, t, x=None, sigma=None):
         best = np.zeros(N, dtype=float)
         last = np.zeros(N, dtype=int)
 
+        # Compute ncp_prior if not defined
+        if self.ncp_prior is None:
+            self.compute_ncp_prior(N)
```

Comment:
One problem I see with this: if a user re-fits the object on a new dataset, they would probably expect ``ncp_prior`` to be appropriately re-computed for the new data. The way this is structured, once the prior is computed for given data, that prior continues to be used.

I think I would keep ``self.ncp_prior`` as the value passed by the user (i.e. ``None`` is possible) and write, e.g. ``ncp_prior = self.compute_ncp_prior(N)``, and use this local version below.